bench/datasets/swe_bench.py

```
@@ -0,0 +1,373 @@
+        try:
+            _git(["apply", "--allow-empty", str(patch_file)], cwd=dest)
+        except subprocess.CalledProcessError as exc:
+            # Some test_patches need 3-way merge; retry with -3.
+            res = subprocess.run(
+                ["git", "apply", "--3way", str(patch_file)],
+                cwd=dest, capture_output=True, text=True,
+            )
+            if res.returncode != 0:
+                raise RuntimeError(
+                    f"failed to apply test_patch for {inst.instance_id}: {exc.stderr}"
+                )
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)